charmhelpers/contrib/storage/linux/ceph.py

```diff
@@ -1713,6 +1718,7 @@ def add_op(self, op):
         """
         if op not in self.ops:
             self.ops.append(op)
+            self.request_id = hashlib.sha1(json.dumps(self.ops, sort_keys=True).encode()).hexdigest()
```

Comment:
Much like my other comment, I think it's worthy moving this to a function, since it's used more than once, and it makes testing it easier.